python/src/linalg.cpp

```diff
@@ -260,4 +260,33 @@ void init_linalg(nb::module_& parent_module) {
         Returns:
             array: ``ainv`` such that ``dot(a, ainv) = dot(ainv, a) = eye(a.shape[0])``
       )pbdoc");
+  m.def(
+      "cholesky",
+      &cholesky,
+      "a"_a,
+      "upper"_a = false,
+      nb::kw_only(),
+      "stream"_a = nb::none(),
+      nb::sig(
+          "def cholesky(a: array, upper: bool = False, *, stream: Union[None, Stream, Device] = None) -> array"),
+      R"pbdoc(
+        Compute the Cholesky decomposition of a real symmetric positive semi-definite matrix.
+
+        This function supports arrays with at least 2 dimensions. When the input
+        has more than two dimensions, the Cholesky decomposition is computed for each matrix
+        in the last two dimensions of ``a``.
+
+        If the input matrix is not symmetric positive semi-definite, behaviour is undefined.
```

Comment:
It currently crashes. Maybe undefined is indeed better. We could just not throw in that case.

Reply:
Yes right, it indeed crashes when matrix is not positive semi-definite. The behaviour I wanted was:

- raise when the matrix is not positive semidefinite, since it is directly detected by Lapack
- undefined if the matrix is not symmetric, since adding a check will come an unnecessary overhead.

Maybe I should specify it in the doc, or do you still prefer not raising at all?

Reply:
I changed it to do nothing and added a comment to add some error checking if we can throw from the implementation in the future. Otherwise, potentially crashing the program in an unrecoverable way based on the values of the input matrix seems an extremely bad idea to me.